cymetric/filters.py

```diff
@@ -0,0 +1,247 @@
+import pandas as pd
+import numpy as np
+
+try:
+    from graphviz import Digraph
+    HAVE_GRAPHVIZ = True
+except ImportError:
+    HAVE_GRAPHVIZ = False
+
+try:
+    from pyne import data
+    import pyne.enrichment as enr
+    from pyne import nucname
+    HAVE_PYNE = True
+except ImportError:
+    HAVE_PYNE = False
+
+
+import cymetric as cym
+from cymetric import tools
+from cymetric.tools import format_nuc
+from cymetric.tools import reduce
+from cymetric.tools import merge
+from cymetric.tools import add_missing_time_step
+
+
+def get_transaction_df(evaler, senders=(), receivers=(), commodities=()):
+    """
+    Filter the Transaction Data Frame on specific sending facility and
+    receving facility.
+
+    Parameters
+    ----------
+    evaler : evaler
+    senders :  of the sending facility
+    receivers :  of the receiving facility
+    commodities :  of the commodity exchanged
+    """
+
+    # initiate evaluation
+    trans = evaler.eval('Transactions')
+    agents = evaler.eval('AgentEntry')
+
+    rec_agent = agents.rename(index=str, columns={'AgentId': 'ReceiverId'})
+    if len(receivers) != 0:
+        rec_agent = rec_agent[rec_agent['Prototype'].isin(receivers)]
+
+    send_agent = agents.rename(index=str, columns={'AgentId': 'SenderId'})
+    if len(senders) != 0:
+        send_agent = send_agent[send_agent['Prototype'].isin(senders)]
+
+    # check if sender and receiver exist
+    if rec_agent.empty or send_agent.empty:
+        return None
+
+    # Clean Transation PDF
+    rdc_ = []
```

Comment:
non-blocking, but it is extremely uncommon to use a trailing underscore in Python variable names.